### PR TITLE
Make sure all joins / rejoins end up reflected in admin participant views

### PR DESCRIPTION
**Pull request by mikewillems**
## What is in this PR?

This PR makes user tracking logic use the same responsive techniques and source of truth for "who is currently present" for joins as for drops. Non-hosted events already did this via the backend estimate. This also adds an on-demand presence refresh so an admin viewers local doc isn't waiting on the 20s heartbeat timer when they open a user mgmt drawer / tab (either pre or during breakouts).

## Changes in the codebase

- `live_meeting_provider.dart`: Add `forcePresenceHeartbeat()`, which writes the same presence heartbeat the periodic 20s timer already writes, callable on demand instead of waiting for the timer. Only writes the local user's own `event-participants` doc.
- `admin_panel.dart`: Hosted-event participant count and roster now come from `EventProvider.presentParticipants` (Firestore presence) instead of `LiveMeetingProvider.meetingProviderParticipants` (Agora-derived). Replaced main function of `_buildMeetingProviderParticipantEntry` with `_buildPresenceParticipantEntry` to first look up which participants are actually present then only after that verify their Agora presence to check ability to mute. Edited `initState` to call `forcePresenceHeartbeat()`.
- `breakout_rooms_dialog.dart`: Already read presence correctly, but I added the same `forcePresenceHeartbeat()` call in `initState`.
--- PLUS REFACTOR FOR MAXPARTICIPANTS CONSISTENCY IN RESPONSE TO TEST FAILURE
- event.dart - new Event.effectiveMaxParticipants getter (single source of truth for the "no value set yet" default), plus lowered the hostless default limit from 10,000 to 1,000.
- edit_event_presenter.dart - edited updateEventType/getHostedMaxParticipantsForUi to use the new getter and only clamp an existing explicit values instead of always stamping a hardcoded default (which was inconsistent and also made the tests break under an earlier attempted fix for the null value problem - the right move was actually making the logic consistent not the quick fix of making the test expect the inconsistent behavior).
- event_info.dart - modified hosted "is event full" check to use the getter instead of its own ?? 0 fallback (which was a latent bug: a hosted event with unset maxParticipants would always read as full/0-capacity).
create_event_dialog_model.dart - again, use getter instead of writing in a value for maxParticipants.

## Changes outside the codebase

None

### Test the participant counts

1. Join a hosted meeting as host/admin on one browser, and as a participant on a second browser.
2. Open the admin panel's Participants list. Confirm count and roster include the second participant.
3. Have the second participant leave, then rejoin. Confirm the count/roster update on both departure and rejoin without needing to refresh.
4. Open the "Breakouts" dialog and confirm its "Current Participants" number matches the admin panel's count.
5. Confirm mute-related controls still show up for participants whose Agora session has been resolved, and are simply absent (not error) for any participant who hasn't connected to Agora yet.

### Test the participant limit defaults
1. Create a hosted event and don't touch the participant limit - confirm it doesn't immediately show as "full" with 0 registrants (regression check for the old ?? 0 bug).
2. Create a new hostless event and verify that registration/join cap is 1k, not 10k.
3. In Edit Event, switch an event with a custom hostless limit (e.g. 500) to "Hosted" and confirm it clamps into the hosted range (2-50) rather than carrying over the raw hostless value.
4. Switch event type back and forth without setting a custom limit and confirm no unintended value gets permanently written until the user actually adjusts the slider.

## PR Checklist

- N/A: Where applicable, I have added localization (l10n) entries to my feature for user-facing text.
- N/A: For new Cloud Functions, I have added the function to `function-mapping.json`. 

## Additional information

There's another pre-existing bug with `GetUserIdFromAgoraId` failing in `agora_room.dart` -> `onUserJoined`. Didn't fix it here, but it no longer affects participant counts since those no longer depend on Agora. It may still be worth hardening with a retry / backoff and error logging since `meetingProviderParticipants` is still used for per-participant mute controls. Haven't tested that thoroughly, but I think that's out of scope.

**AI tools used (if applicable)**: 

Today is not that day!

### CODE REVIEW

File: client/lib/features/events/features/live_meeting/features/admin_panel/presentation/widgets/admin_panel.dart [hunk 1]
```diff
@@ -206,10 +228,9 @@ class _AdminPanelState extends State<AdminPanel> {
         ),
         Expanded(
           child: ListView(
-            children: _provider.meetingProviderParticipants
-                    ?.map(_buildMeetingProviderParticipantEntry)
-                    .toList() ??
-                [],
+            children: presentParticipants
+                .map(_buildPresenceParticipantEntry)
+                .toList(),
```
[the remark this answers is not in the document]
Comment (johnnycrich) [reply client/lib/features/events/features/live_meeting/features/admin_panel/presentation/widgets/admin_panel.dart:222]: @mikewillems I don't agree that this is as crucial as Copilot is suggesting, but I suppose a quick fix would just be to do:

 `onPressed: _provider.meetingProviderParticipants == null
              ? null
              : () => _providerRead.muteAllParticipants(),` 
              
...for the mute all action.

File: client/lib/features/events/features/live_meeting/features/admin_panel/presentation/widgets/admin_panel.dart [hunk 2]
```diff
@@ -75,28 +87,32 @@ class _AdminPanelState extends State<AdminPanel> {
...
             ),
           ),
           Spacer(),
-          if (!local)
+          if (participant.id != currentUserId)
             _ParticipantMenu(
-              kickedUserId: participant.userId,
-              providerParticipant: participant,
+              kickedUserId: participant.id,
+              providerParticipant: providerParticipant,
```
[the remark this answers is not in the document]
Comment (johnnycrich) [reply client/lib/features/events/features/live_meeting/features/admin_panel/presentation/widgets/admin_panel.dart:107]: @mikewillems This is pretty tiny for a very time-limited potential issue, but I think the suggested fix would work.

**Review by johnnycrich — APPROVED** [review]
@mikewillems These are nice, atomic fixes! Just see the couple comments about the copilot notes that I think have merit, otherwise I think this is good to go!

File: client/lib/features/events/features/live_meeting/features/admin_panel/presentation/widgets/admin_panel.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (mikewillems) [reply client/lib/features/events/features/live_meeting/features/admin_panel/presentation/widgets/admin_panel.dart:224]: Good call, I'll work it in!

File: client/lib/features/events/features/live_meeting/features/admin_panel/presentation/widgets/admin_panel.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (mikewillems) [reply client/lib/features/events/features/live_meeting/features/admin_panel/presentation/widgets/admin_panel.dart:107]: I'm not sure that would ever be able to come up; I'll look into it.
